### PR TITLE
Round UI coordinates after scaling

### LINKED ISSUE #9754: Pixel gap between adjacent UI grid cells
## Bevy version
11.2

## Relevant system information
```
INFO winit::platform_impl::platform::x11::window: Guessed window scale factor: 1.1666666666666667
INFO bevy_render::renderer: AdapterInfo { name: "NVIDIA GeForce GTX 970", vendor: 4318, device: 5058, device_type: DiscreteGpu, driver: "NVIDIA", driver_info: "535.98", backend: Vulkan }
```

## What you did

```rust
use bevy::prelude::*;

fn main() {
    App::new()
        .add_plugins(DefaultPlugins)
        .add_systems(Startup, setup)
        .run();
}

fn setup(mut commands: Commands) {
    commands.spawn(Camera2dBundle::default());

    commands
        .spawn(NodeBundle {
            style: Style {
                display: Display::Grid,
                grid_template_columns: RepeatedGridTrack::min_content(2),
                margin: UiRect::all(Val::Px(4.0)),
                ..Default::default()
            },
            background_color: BackgroundColor(Color::PINK),
            ..Default::default()
        })
        .with_children(|commands| {
            for color in [Color::GRAY, Color::DARK_GRAY] {
                commands.spawn(NodeBundle {
                    style: Style {
                        display: Display::Grid,
                        width: Val::Px(160.),
                        height: Val::Px(160.),
                        ..Default::default()
                    },
                    background_color: BackgroundColor(color),
                    ..Default::default()
                });
            }
        });
}
```

## What went wrong

The vertical pink line seems to be related to the margin, as it doesn't appear with a zero margin. But varying values of width and height for the cells also make the gap disappear. Maybe related to #7712, although not a regression (I tested the sample on 11.2 and it works).

grid_gap

### PR DESCRIPTION
**Pull request by ickshonpe**
# Objective

Fixes #9754

## Solution

Don't round UI coordinates until they've been multiplied by the inverse scale factor.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

**Comment by nicoburns** [comment]
I think this is probably an improvement in that it brings the rounding later in the process (thus reducing float error). But it looks like this is now rounding in logical coordinates, whereas I think ideally it needs to be done in physical coordinates but after Transform has been applied. Not sure how feasible this is.

**Comment by ickshonpe** [comment]
> I think this is probably an improvement in that it brings the rounding later in the process (thus reducing float error). But it looks like this is now rounding in logical coordinates, whereas I think ideally it needs to be done in physical coordinates but after Transform has been applied. Not sure how feasible this is.

I tried a couple of approaches and this one seemed to be the most accurate. By after Transform has been applied do you mean after the GlobalTransform has been computed?

**Review by mockersf — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
@ickshonpe I'll merge this once you fix merge conflicts (I love import merge conflicts...).

**Pull request merged**
